src/movement_controls.py

```diff
@@ -1,12 +1,17 @@
 from __future__ import annotations
 
 import math
+from typing import TYPE_CHECKING
 
 import arcade
 
 import constants
 from player_input import PlayerInput
 
+# This allows a circular import only for the purposes of type hints
+if TYPE_CHECKING:
+    from vehicle import Vehicle
```

Comment:
What is the goal in locking the vehicle to the movement controls?

The movement only needs the [not added yet] collider sprite. At least for as long as the placeholder map is still in use. By setting the reference rather than passing it as a parameter I only see it creating an additional pain point for changing the vehicle the player controls[at runtime] or making code changes

Reply:
The way I understood it, the vehicle owns it's movement controls, so it made sense for the movement controls to have a reference to the vehicle. I figured if the player got a new vehicle somehow, it would create it's own instance of movement controls. Maybe that wouldn't work so well if the player somehow got control of a pre-existing vehicle. Want me to change it back to passing the vehicle in each method?

Reply:
Alternatively the movement controls could be owned by the player (with a reference to the player) and act on whatever vehicle the player currently controls. Let me know your thoughts

Reply:
Fixed: I put it back to the way it was with the vehicle being passed in the methods instead of the initializer

Reply:
what is the design behind passing 'vehicle' instead of just the sprite?

If I am understanding these changes it is:
1: Adding the 'vehicle' to player to set different cars
and
B: Moving all the moving parts [sprites and input] to vehicle

if B there needs to be a decision about the structure of who/what/how a player is mechanically

Reply:
The main purpose of passing `vehicle` rather than the sprite is that the vehicle has weapons sprites on the hood that it needs to update whenever the vehicle is moved. By telling the vehicle to move and subsequently having it move it's own sprite, it can take care of the weapons sprites as well.

Happy to go into if further on discord if you're not convinced or have suggestions